src/lib.rs

```diff
@@ -904,7 +930,7 @@ impl SeedableRng for StdRng {
 /// This will seed the generator with randomness from thread_rng.
 #[cfg(feature="std")]
 pub fn weak_rng() -> XorShiftRng {
-    thread_rng().gen()
+    XorShiftRng::new().unwrap()
```

Comment:
Not great to unwrap here, but hopefully we can replace `weak_rng` with a `SmallRng` wrapper type soon, and let user code use it with `NewSeeded`.

Reply:
I think `thread_rng` is the better way to seed new `weak_rng`s anyway? It should perform much better if multiple weak RNGs are created, so use `from_rng`? Unfortunately you still need some error handling to retry a few times in case `XorshiftRng::from_rng` fails with seed 0.

Reply:
I was not sure it is better. If a library already makes use of `thread_rng` it will be faster. But if it never uses `thread_rng` the first use of `weak_rng` will be slower, because `thread_rng` is far more costly to initialize.

I don't care much either way, this mirrors `NewSeeded`.

Reply:
Only the first use; I think it would be nice to make `weak_rng` fairly fast average-case.